### ISSUE TITLE
 Player InterruptedException (Connection aborted)

### ISSUE THREAD

**Issue by patolax** [ryanheise/just_audio#1170]
**Which API doesn't behave as documented, and how does it misbehave?**
play() method throws Player InterruptedException (Connection aborted)  error when called multiple times asynchronously. 

**Minimal reproduction project**
Provide a link here using one of two options:
1. [ Fork this repository and modify the example to reproduce the bug, then provide a link here.](https://github.com/patolax/just_audio/tree/main/example)

**To Reproduce (i.e. user steps, not code)**
1. Go to 'Main page'
2. Click on 'Play Effects'
3. See the 'Player InterruptedException' error in the console

**Error messages**

```
Player InterruptedException (Connection aborted)
```

**Expected behavior**
No such error or way to handle this

**Screenshots**
![image](https://github.com/ryanheise/just_audio/assets/29444156/97e68906-e739-4466-b920-4bce93d5fcc6)

**Desktop (please complete the following information):**
 - Microsoft Windows [Version 10.0.19045.3930],

**Smartphone (please complete the following information):**
 - Nokia g20 [android 13]

**Flutter SDK version**
```
√] Flutter (Channel stable, 3.13.6, on Microsoft Windows [Version 10.0.19045.3930], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Build Tools 2019 16.11.18)
[!] Android Studio (version 2021.2)
X Unable to determine bundled Java version.
[√] Android Studio (version 2022.3)
[√] VS Code, 64-bit edition (version 1.85.1)
[√] Connected device (4 available)
[√] Network resources
```

**Additional context**
```static Future playAll() async {
   List li = await Future.wait([
    playSfx(), 
    playSfx(), 
    playSfx(), 
    playSfx(), 
  ]);
  }

  static Future<void> playSfx({double volume = 1}) async {
    try {
      
        final duration = await _player.setUrl("asset:///audio/nature.mp3");                 // Schemes: (https: | file: | asset: )
        _player.play();

    } catch (e) {
      debugPrint('Player error ${e.toString()}');
    }
  }```

**Comment by ryanheise** [comment]
I'm having trouble reading the commit log / diffs for your fork, so it is very difficult to understand what minimal changes you have made.

I'm also confused by your first section in the bug report which is supposed to qualify this as a bug by explaining how the actual behaviour is different from the documented behaviour. But the behaviour you describe doesn't contradict anything, it seems to be functioning exactly according to the documentation.

 Since it was not qualified as a bug, I'll close the issue. If you disagree, please fill in a correct bug report that qualifies it as a bug and also links to a fork with "minimal" changes as visible in the commit diffs.

**Comment by patolax** [comment]
The additional context section shows the only changes I made.
This is that code
https://github.com/patolax/just_audio/blob/main/example/lib/myplayer.dart
I call PlayAll from the main page. That is it. It will throw the exception.

I found out this error is thrown when the player is trying to play a new file while one sound has been already played. I would like to avoid this behavior. I tried following the code but the behaviour is so random.

`if (_player.processingState == ProcessingState.completed ||
          _player.processingState == ProcessingState.idle) {
        await _player.setAsset('assets/audio/$file');
        await _player.setVolume(volume);
        await _player.play();
      }`
      
  Any recommendations as to how to handle this? this question is asked in many places without a real answer.
